bindings/tbdex_wasm/src/js.rs

```diff
@@ -0,0 +1,50 @@
+use js_sys::{Array, Map, Object, Reflect};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsValue;
+
+fn convert_to_object_recursive(value: &JsValue) -> Result<JsValue, JsValue> {
```

Comment:
This replaces the `mapToObject()` which was removed in the typescript code, I figure it's a bit more robust to move this in to the rust layer. But also, I suspect in the future we may revamp our bindings code with the intent of improving maintainability, in which case this may go away.